more_itertools/recipes.py

```diff
@@ -1075,3 +1077,89 @@ def totient(n):
     for prime in set(factor(n)):
         n -= n // prime
     return n
+
+
+# Miller–Rabin primality test: Testing against small sets of bases
+# https://w.wiki/CGwr
+_perfect_tests = [
+    (2047, (2,)),
+    (9080191, (31, 73)),
+    (4_759_123_141, (2, 7, 61)),
+    (1122004669633, (2, 13, 23, 1662803)),
+    (2152302898747, (2, 3, 5, 7, 11)),
+    (3474749660383, (2, 3, 5, 7, 11, 13)),
+    (341550071728321, (2, 3, 5, 7, 11, 13, 17)),
+    (3825123056546413051, (2, 3, 5, 7, 11, 13, 17, 19, 23)),
```

Comment:
Perhaps replace this one with:

`(18446744073709551616, (2, 325, 9375, 28178, 450775, 9780504, 1795265022)),`

That covers all of `2 ** 64` with only seven rounds.